lib/pbench/cli/server/shell.py

```diff
@@ -158,34 +134,32 @@ def keycloak_connection(config: PbenchServerConfig, logger: Logger) -> int:
 
 
 def main():
-    """The Pbench Server is always executed in a containerized environment where
-    the pbench server configuration is in a fixed location.
+    """Setup of the Gunicorn Pbench Server Flask application.
 
     If an error is encountered during setup, system exit is invoked with an
     exit code of 1.  Otherwise, the exit code of the gunicorn subprocess is
     used.
     """
-    os.environ[
-        "_PBENCH_SERVER_CONFIG"
-    ] = "/opt/pbench-server/lib/config/pbench-server.cfg"
     try:
         server_config = get_server_config()
     except (ConfigFileNotSpecified, BadConfig) as e:
         print(e)
         sys.exit(1)
     logger = get_pbench_logger(PROG, server_config)
-    ret_val = keycloak_connection(config=server_config, logger=logger)
-    if ret_val != 0:
-        sys.exit(ret_val)
     if site.ENABLE_USER_SITE:
         find_the_unicorn(logger)
     try:
         host = str(server_config.get("pbench-server", "bind_host"))
         port = str(server_config.get("pbench-server", "bind_port"))
-        db_uri = str(server_config.get("database", "db_uri"))
+        db_uri = str(server_config.get("database", "uri"))
+        db_wait_timeout = str(server_config.get("database", "wait_timeout"))
         workers = str(server_config.get("pbench-server", "workers"))
         worker_timeout = str(server_config.get("pbench-server", "worker_timeout"))
-        crontab_dir = server_config.get("pbench-server", "crontab-dir")
+        oidc_server = server_config.get(
+            "authentication",
+            "internal_server_url",
+            fallback=server_config.get("authentication", "server_url"),
+        )
     except (NoOptionError, NoSectionError):
         logger.exception("Error fetching required configuration")
```

Comment:
Given the number of possibilities for failure, perhaps we should provide better feedback than this (yeah, the traceback probably has everything, but we don't really need/want a traceback here, especially if the message supplied the information in a nice format).  I think that those exception objects provide a nice formatting automagically:
```
    except (NoOptionError, NoSectionError) as exc:
        logger.error("Error fetching required configuration: {}", str(exc))
```